variables.tf

```diff
@@ -37,30 +37,19 @@ variable "region" {
     error_message = "The specified region is not valid, supported regions are: us-south (Dallas), eu-gb (London), eu-de (Frankfurt), au-syd (Sydney), eu-es (Madrid)"
   }
 }
-
-variable "kms_region" {
-  type        = string
-  description = "The region where KMS instance exists if using KMS encryption."
-  default     = "us-south"
-}
-
-variable "kms_endpoint" {
-  description = "The KMS endpoint to use when configuring KMS encryption. Must be private or public."
+variable "kms_endpoint_url" {
+  description = "The KMS endpoint URL to use when configuring KMS encryption. HPCS endpoint URL format- https://api.private.<REGION>.hs-crypto.cloud.ibm.com:<port> and KP endpoint URL format- https://<REGION>.kms.cloud.ibm.com"
   type        = string
-  default     = "public"
-  validation {
-    condition     = contains(["public", "private"], var.kms_endpoint)
-    error_message = "Valid values for kms_endpoint are 'public' or 'private'."
-  }
+  default     = null
 }
 
 variable "service_endpoints" {
   type        = string
   description = "Specify whether you want to enable the public or both public and private service endpoints. Supported values are 'public' or 'public-and-private'."
   default     = "public-and-private"
   validation {
-    condition     = contains(["public", "public-and-private"], var.service_endpoints)
-    error_message = "The specified service endpoint is not a valid selection! Supported options are: public or public-and-private."
+    condition     = contains(["public", "private", "public-and-private"], var.service_endpoints)
```

Comment:
Are you sure `private` is supported? I don't see it as an available option in the UI:

![image](https://github.com/terraform-ibm-modules/terraform-ibm-event-notifications/assets/29608224/109cbabe-ea79-4601-8c86-833db9a88767)

Reply:
- @ocofaigh I had a discussion with @jojustin on this before and it was discussed that though we set endpoint as 'private', after the deployement the endpoint will be set as "public and private" only for the respective service.
- And we must set `private` for fscloud profile.

Reply:
So are you saying that terraform allows you to pass `private` as a value, however you end up with an instance that actually has private and public enabled? 
If so, I don't think we should allow `private` to be passed here. It gives the impression that the instance cannot be accessed by a public endpoint when infact it can. If the service doesn't support private only, then our module should not support it

Reply:
Ok, looks like all endpoints are supported- in [docs](https://cloud.ibm.com/docs/event-notifications?topic=event-notifications-en-regions-endpoints) it is mentioned both public and private and I tested with all.
-  I just tested this again with `private` endpoint and I see the instance only has `private` endpoint.
   <img width="1697" alt="image" src="https://github.com/terraform-ibm-modules/terraform-ibm-event-notifications/assets/4889052/152d0eac-05eb-4879-9c6a-fe0db96806b7">

- Tested with `public-and-private`
   <img width="1689" alt="image" src="https://github.com/terraform-ibm-modules/terraform-ibm-event-notifications/assets/4889052/beb022bc-3430-43ab-af38-6993f2c555b7">

- Test with `public`
  <img width="1679" alt="image" src="https://github.com/terraform-ibm-modules/terraform-ibm-event-notifications/assets/4889052/bfa26411-4dfe-4fc6-9790-a2a2712c87b9">

Reply:
@Ak-sky can you follow up with EN team to find out why its not exposed in the UI?

Reply:
Posted the [message](https://ibm-cloudplatform.slack.com/archives/C0267256J4F/p1709187937546029) on the slack channel.

Reply:
Private endpoint not supported, changing back to public-and-private endpoint